Sources/SwiftDocC/Checker/Checkers/TutorialDirectiveMisuseChecker.swift

```diff
@@ -0,0 +1,62 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import Markdown
+
+public struct TutorialDirectiveMisuseChecker: Checker {
+    public var problems = [Problem]()
+
+    private var sourceFile: URL?
+
+    public init(sourceFile: URL?) {
+        self.sourceFile = sourceFile
+    }
+
+    public mutating func visitBlockDirective(_ blockDirective: BlockDirective) {
+        guard let blockRange = blockDirective.range else { return }
+        
+        let diagnostic: Diagnostic
+        let solutions: [Solution]
+
+        switch blockDirective.name {
+        case ImageMedia.directiveName:
+            diagnostic = Diagnostic(
+                source: sourceFile,
+                severity: .warning,
+                range: blockRange,
+                identifier: "org.swift.docc.TutorialDirectiveMisuse",
+                summary: #""@\#(blockDirective.name)" directive is not support on normal markdown file."#,
```

Comment:
It would be good to use a similar style to this diagnostic we already have: https://github.com/apple/swift-docc/blob/main/Sources/SwiftDocC/Model/DocumentationNode.swift#L396, e.g., "Directive 'Image' is not supported in…".

Rather than "normal markdown file", the DocC naming is either "article" or "documentation extension". Can we know easily what kind of markdown file this Checker being run on so that we show the right text?

Reply:
Gotcha. Yes I think it would be good to run this on documentation extensions as well.